modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/upgrade/v4_3_0/test/KaleoLogUpgradeProcessTest.java

```diff
@@ -0,0 +1,173 @@
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.portal.workflow.kaleo.definition.LogType;
+import com.liferay.portal.workflow.kaleo.definition.Task;
+import com.liferay.portal.workflow.kaleo.model.KaleoInstance;
+import com.liferay.portal.workflow.kaleo.model.KaleoInstanceToken;
+import com.liferay.portal.workflow.kaleo.model.KaleoLog;
+import com.liferay.portal.workflow.kaleo.model.KaleoNode;
+import com.liferay.portal.workflow.kaleo.runtime.util.WorkflowContextUtil;
+import com.liferay.portal.workflow.kaleo.service.KaleoInstanceLocalService;
+import com.liferay.portal.workflow.kaleo.service.KaleoInstanceTokenLocalService;
+import com.liferay.portal.workflow.kaleo.service.KaleoLogLocalService;
+import com.liferay.portal.workflow.kaleo.service.KaleoNodeLocalService;
+
+import java.io.Serializable;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Pedro Leite
+ */
+@RunWith(Arquillian.class)
+public class KaleoLogUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext();
+
+		_kaleoInstance = _kaleoInstanceLocalService.addKaleoInstance(
+			1, 1, RandomTestUtil.randomString(), 1,
+			HashMapBuilder.<String, Serializable>put(
+				WorkflowConstants.CONTEXT_ENTRY_CLASS_NAME,
+				BlogsEntry.class.getName()
+			).put(
+				WorkflowConstants.CONTEXT_SERVICE_CONTEXT, serviceContext
+			).build(),
+			serviceContext);
+
+		KaleoNode kaleoNode = _kaleoNodeLocalService.addKaleoNode(
+			_kaleoInstance.getKaleoDefinitionId(),
+			_kaleoInstance.getKaleoDefinitionVersionId(),
+			new Task(RandomTestUtil.randomString(), StringPool.BLANK),
+			serviceContext);
+
+		KaleoInstanceToken kaleoInstanceToken =
+			_kaleoInstanceTokenLocalService.addKaleoInstanceToken(
+				kaleoNode.getKaleoNodeId(),
+				_kaleoInstance.getKaleoDefinitionId(),
+				_kaleoInstance.getKaleoDefinitionVersionId(),
+				_kaleoInstance.getKaleoInstanceId(), 0,
+				WorkflowContextUtil.convert(
+					_kaleoInstance.getWorkflowContext()),
+				serviceContext);
+
+		_updateKaleoLogType(
+			_kaleoLogLocalService.addInstanceStartKaleoLog(
+				kaleoInstanceToken, serviceContext),
+			"WORKFLOW_INSTANCE_START");
+
+		_updateKaleoLogType(
+			_kaleoLogLocalService.addInstanceEndKaleoLog(
+				kaleoInstanceToken, serviceContext),
+			"WORKFLOW_INSTANCE_END");
```

Comment:
@pedro-oliveira446 can we remove the empty line and sort?

Reply:
The order matters here!